exchange/utils.go

```diff
@@ -39,34 +41,32 @@ func cleanOpenRTBRequests(ctx context.Context,
 
 	requestsByBidder, errs = splitBidRequest(orig, impsByBidder, aliases, usersyncs, blables, labels)
 
-	// Clean PI from bidrequests if not allowed per GDPR
 	gdpr := extractGDPR(orig, usersyncIfAmbiguous)
 	consent := extractConsent(orig)
-
-	// Check if it's an AMP request
 	isAMP := labels.RType == pbsmetrics.ReqTypeAMP
 
-	// Check if COPPA applies for this request
-	var applyCOPPA bool
-	if orig.Regs != nil && orig.Regs.COPPA == 1 {
-		applyCOPPA = true
+	privacyEnforcement := privacy.Enforcement{
+		COPPA: orig.Regs != nil && orig.Regs.COPPA == 1,
+	}
+
+	if enforceCCPA {
+		ccpaPolicy, _ := ccpa.ReadPolicy(orig)
+		privacyEnforcement.CCPA = ccpaPolicy.ShouldEnforce()
 	}
 
 	for bidder, bidReq := range requestsByBidder {
-		var applyGDPR bool
-		// Fixes #820
+
 		if gdpr == 1 {
 			coreBidder := resolveBidder(bidder.String(), aliases)
 
 			var publisherID = labels.PubID
-			if ok, err := gDPR.PersonalInfoAllowed(ctx, coreBidder, publisherID, consent); !ok && err == nil {
-				applyGDPR = true
-			}
+			ok, err := gDPR.PersonalInfoAllowed(ctx, coreBidder, publisherID, consent)
+			privacyEnforcement.GDPR = !ok && err == nil
```

Comment:
I realize this was not in the code even before this PR, but if `gDPR.PersonalInfoAllowed()` throws an error, should we log it somewhere? It seems that in case an `ErrorMalformedConsent` gets created in `gdpr/impl.go`'s line 106, said error won't get logged to `stderr` nor anywhere else.

Reply:
I'm not sure. I'll ask the regulation team at our next meeting for their thoughts. I don't know the context or reasoning for ignoring it.